examples/sphere/shallow_water_dss.jl

```diff
@@ -0,0 +1,94 @@
+ENV["CLIMACORE_DISTRIBUTED"] = "MPI"
+usempi = true
+using LinearAlgebra
+using Colors
+using JLD2
+
+include("../nvtx.jl")
+
+import ClimaCore:
+    Domains,
+    Fields,
+    Geometry,
+    Meshes,
+    Operators,
+    Spaces,
+    Topologies,
+    DataLayouts
+
+using Logging
+if usempi
+    using ClimaComms
+    using ClimaCommsMPI
+else
+    import TerminalLoggers
+end
+
+set_initial_condition(space) =
+    map(Fields.local_geometry_field(space)) do local_geometry
+        coord = local_geometry.coordinates
+        h = 1.0
+        u = Geometry.transform(
+            Geometry.Covariant12Axis(),
+            Geometry.UVVector(1.0, 1.0),
+            local_geometry,
+        )
+        return (h = h, u = u)
+    end
+
+function shallow_water_dss_profiler(usempi::Bool, ::Type{FT}) where {FT}
+    context = ClimaCommsMPI.MPICommsContext()
+    pid, nprocs = ClimaComms.init(context)
+    iamroot = ClimaComms.iamroot(context)
+    # log output only from root process
+    logger_stream = iamroot ? stderr : devnull
+    prev_logger = global_logger(ConsoleLogger(logger_stream, Logging.Info))
+    atexit() do
+        global_logger(prev_logger)
+    end
+
+    if iamroot
+        println("running distributed simulation using $nprocs processes")
+    end
+    # Set up discretization
+    ne = 9 # the rossby_haurwitz test case's initial state has a singularity at the pole. We avoid it by using odd number of elements
+    Nq = 4
+    R = FT(6.37122e6) # radius of earth
+    domain = Domains.SphereDomain(R)
+    mesh = Meshes.EquiangularCubedSphere(domain, ne)
+    quad = Spaces.Quadratures.GLL{Nq}()
+    grid_topology = Topologies.DistributedTopology2D(context, mesh)
+    global_grid_topology = Topologies.Topology2D(mesh)
+    space = Spaces.SpectralElementSpace2D(grid_topology, quad)
+    global_space = Spaces.SpectralElementSpace2D(global_grid_topology, quad)
+    Y = set_initial_condition(space)
+    ghost_buffer = usempi ? Spaces.create_ghost_buffer(Y) : nothing
+    # Set up RHS function
+    dYdt = similar(Y)
+    # compile
+    @nvtx "dss-compiling" color = colorant"green" begin
+        Spaces.weighted_dss_start!(dYdt, ghost_buffer)
+        Spaces.weighted_dss_internal!(dYdt, ghost_buffer)
+        Spaces.weighted_dss_ghost!(dYdt, ghost_buffer)
+    end
+
+    # profile
+    walltime = @elapsed begin
+        for i in 1:100
+            @nvtx "dss-loop" color = colorant"green" begin
+                Spaces.weighted_dss_start!(dYdt, ghost_buffer)
+                Spaces.weighted_dss_internal!(dYdt, ghost_buffer)
+                Spaces.weighted_dss_ghost!(dYdt, ghost_buffer)
```

Comment:
It would be useful for comparison to have an option to do it without the actual DSS operations. i.e. just
```julia
fill_send_buffer!(topology, data, ghost_buffer)
ClimaComms.start(ghost_buffer.graph_context)
ClimaComms.finish(ghost_buffer.graph_context)
```